### DISCUSSION TITLE
`vega_datasets` source

### DISCUSSION THREAD

**Discussion by ChristopherDavisUCI** [vega/altair#3150] [General]
Would there be a reliable way to transition from using https://github.com/altair-viz/vega_datasets as the default source of example datasets to instead https://github.com/vega/vega-datasets ?  Or, if that's not realistic, would there be a way to synchronize the data in the altair-viz version with the data in the vega version?  (As far as I can tell, they're not synchronized.)

I could imagine having no local datasets and having functionality like `alt.vega_datasets.cars()` which downloads the cars dataset from https://github.com/vega/vega-datasets/blob/main/data/cars.json 

I don't know at all about the tradeoffs involved, it just feels like this approach might remove one step of maintenance.

Any thoughts on that?

**Comment by mattijn** [comment]
I like the idea. I remember it took me a while before realizing they are not the same repository. 

The goal of `vega_datasets` was to make the datasets used by the Vega project easily accessible in Python in the form of a Pandas dataframe.

Since it is also used as teaching material, I can imagine we also like to provide a wider accessibility to choose which dataframe library a user prefers.

An interesting change with the past is that the Vega datasets are currently versioned by GitHub releases. Maybe that provides a new entrypoint for Altair.

Since this will change all examples in the documentation, we probably might think this through a bit how we can support users that are bound to older versions.

**Comment by dangotbanned** [comment]
I'm quite interested in this, so I opened https://github.com/vega/altair/pull/3631

I'd noticed a lot of examples using `vega_datasets` purely for the `url`, and that part of the package is really quite short.

**Reply by dangotbanned** [reply]
@ChristopherDavisUCI, @mattijn I feel like I've got this to good place now for feedback.

- #3631

I've got a lot of docs and tests already, which I'm hoping give a feel for the current API.
Any thoughts large or small would be super appreciated

**Reply by mattijn** [reply]
Will defining the dataset by URL goes through "Loader"?

**Comment by dangotbanned** [comment]
@ChristopherDavisUCI I hope it is okay with you, I've renamed this discussion to help with visibility.

I can revert if not, but I stumbled upon this only after searching for previous `vega_datasets` threads. 
I was already working on #3631 locally, but thought it would be helpful to share after discovering this

**Reply by ChristopherDavisUCI** [reply]
Hi @dangotbanned Sure of course, thanks for looking into this!  (I admit it is not very fresh in my memory what the different considerations were.)

**Comment by ChristopherDavisUCI** [comment]
Thank you for your work on this @dangotbanned !  I'm sorry that I haven't
had a chance to look into this yet, but it is still on my radar.  Please
don't hesitate to send pings.  -Chris
